hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFileOperationCost.java

```diff
@@ -176,6 +176,77 @@ public void testCostOfListFilesOnNonExistingDir() throws Throwable {
             .plus(GET_FILE_STATUS_FNFE)));
   }
 
+  @Test
+  public void testCostOfListStatusOnFile() throws Throwable {
+    describe("Performing listStatus() on a file");
+    Path file = path(getMethodName() + ".txt");
+    S3AFileSystem fs = getFileSystem();
+    touch(fs, file);
+    verifyMetrics(() ->
+            fs.listStatus(file),
+            whenRaw(LIST_STATUS_LIST_OP
+                    .plus(GET_FILE_STATUS_ON_FILE)),
+            whenAuthoritative(LIST_STATUS_LIST_OP),
+            whenNonauth(LIST_STATUS_LIST_OP));
+//    resetMetricDiffs();
+//    fs.listStatus(file);
+//    if (!fs.hasMetadataStore()) {
+//      metadataRequests.assertDiffEquals(1);
+//    }
+//    listRequests.assertDiffEquals(1);
+  }
+
+  @Test
+  public void testCostOfListStatusOnEmptyDir() throws Throwable {
+    describe("Performing listStatus() on an empty dir");
+    Path dir = path(getMethodName());
+    S3AFileSystem fs = getFileSystem();
+    fs.mkdirs(dir);
+    verifyMetrics(() ->
+            fs.listStatus(dir),
+            whenRaw(LIST_STATUS_LIST_OP
```

Comment:
nice to see you picking up the new metrics validation I think it makes a big difference. If there are ways it can be improved, feel free to do so.

Reply:
Sure Thanks :)